app/Http/Controllers/ApiPhotosController.php

```diff
@@ -239,11 +256,44 @@ public function addTags (AddTagsApiRequest $request)
             'request' => $request->all()
         ]);
 
-        dispatch (new AddTags($request->all(), $user->id));
+        dispatch (new AddTags($user->id, $photo->id, $request->litter));
 
         return ['success' => true, 'msg' => 'dispatched'];
     }
 
+    /**
+     * Upload Photo together with its tags
+     *
+     * @param Request $request
+     * @return array
+     * @throws GuzzleException
+     */
+    public function storeWithTags (Request $request) :array
+    {
+        $request->validate([
+            'photo' => 'required|mimes:jpg,png,jpeg,heic',
+            'lat' => 'required|numeric',
+            'lon' => 'required|numeric',
+            'date' => 'required',
+            'tags' => 'required|array'
+        ]);
```

Comment:
Hmm, true, we should make it easier to get fewer errors, but at least we should inform them what's going wrong. If we remove the validation errors they'll get a generic message, because a server error will happen. Instead, with validations in place, we can inform them what's wrong. We should simply use the error messages returned from the backend and show them on the app, but the validations are an absolutely necessary layer.